src/sentry/sentry_metrics/querying/visitors/query_expression.py

```diff
@@ -422,6 +424,10 @@ def _extract_unit(self, timeseries: Timeseries) -> str | None:
 
         parsed_mri = parse_mri(timeseries.metric.mri)
         if parsed_mri is not None:
+            if parsed_mri.name.startswith(SPAN_ATTRIBUTE_PREFIX):
+                rule_id = int(parsed_mri.name[len(SPAN_ATTRIBUTE_PREFIX) :])
+                return SpanAttributeExtractionRuleCondition.objects.get(id=rule_id).config.unit
```

Comment:
We would have to handle the lack of the condition by catching the `SpanAttributeExtractionRuleCondition.DoesNotExist` exception.

Reply:
Good point, added the handling